src/App.tsx

```diff
@@ -1,14 +1,74 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoStatus } from './types/TodoStatus';
+import { FilterTodos } from './types/FilterTodos';
+
+const filterTodos:FilterTodos = (todos, query, completedStatus) => {
+  let prepearedTodos = [...todos];
+
+  if (query.trim()) {
+    prepearedTodos = prepearedTodos.filter(todo => {
+      const normalizedTitle = todo.title.toLowerCase();
+      const normalizedQuery = query.toLowerCase().trim();
```

Comment:
for now you transform `query` on each callback call
will be better to move it outside of the `filter` method